client/ownDevice.go

```diff
@@ -31,27 +30,23 @@ func (c *Client) OwnDevice(ctx context.Context, deviceID string, opts ...OwnOpti
 	return c.deviceOwner.OwnDevice(ctx, deviceID, cfg.otmTypes, cfg.discoveryConfiguration, c.ownDeviceWithSigners, cfg.opts...)
 }
 
-func (c *Client) updateCache(ctx context.Context, d *RefDevice, deviceID string) {
-	if d.DeviceID() != deviceID {
-		if c.deviceCache.RemoveDevice(deviceID, d) {
-			for {
-				storedDev, stored := c.deviceCache.TryStoreDevice(d)
-				if stored {
-					break
-				}
-				c.deviceCache.RemoveDevice(storedDev.DeviceID(), storedDev)
-				storedDev.Release(ctx)
-			}
+func (c *Client) updateCache(ctx context.Context, d *core.Device, oldDeviceID string) {
+	if d.DeviceID() != oldDeviceID {
+		exp, ok := c.deviceCache.GetDeviceExpiration(oldDeviceID)
+		if ok && exp.IsZero() {
+			c.deviceCache.UpdateOrStoreDevice(d)
+		} else {
+			c.deviceCache.UpdateOrStoreDeviceWithExpiration(d)
 		}
+		_, _ = c.deviceCache.LoadAndDeleteDevice(ctx, oldDeviceID)
```

Comment:
add comment: we can't close the device connection because only the deviceID has been changed and was updated in the cache.

Reply:
fixed